node/builder.go

```diff
@@ -392,6 +400,23 @@ func StorageMiner(out *api.StorageMiner) Option {
 	)
 }
 
+func LiteModeOverrides(gapi api.GatewayAPI) Option {
```

Comment:
Instead of overriding/unsetting things here, I'd
* Add another flag to settings to indicate lite mode
* In `Online()` have a 3 sections
  * `ApplyIf(isType(repo.FullNode), ...` for full node specific things
  * `ApplyIf(isLite, ...` for lite node specific things
  * `ApplyIf(isFullOrLite, ...` for common stuff